WordPressAuthenticator/Unified Auth/View Related/Site Address/SiteAddressViewModel.swift

```diff
@@ -0,0 +1,136 @@
+import Foundation
+import WordPressKit
+
+struct SiteAddressViewModel {
+    private let isSiteDiscovery: Bool
+    private let xmlrpcFacade: WordPressXMLRPCAPIFacade
+    private unowned let authenticationDelegate: WordPressAuthenticatorDelegate
+    private let blogService: WordPressComBlogService
+    private var loginFields: LoginFields
+
+    private let tracker = AuthenticatorAnalyticsTracker.shared
+
+    init(isSiteDiscovery: Bool,
+         xmlrpcFacade: WordPressXMLRPCAPIFacade,
+         authenticationDelegate: WordPressAuthenticatorDelegate,
+         blogService: WordPressComBlogService,
+         loginFields: LoginFields
+    ) {
+        self.isSiteDiscovery = isSiteDiscovery
+        self.xmlrpcFacade = xmlrpcFacade
+        self.authenticationDelegate = authenticationDelegate
+        self.blogService = blogService
+        self.loginFields = loginFields
+    }
+
+    enum GuessXMLRPCURLResult: Equatable {
+        case success
+        case error(NSError, String?)
+        case troubleshootSite
+        case customUI(UIViewController)
+    }
+
+    func guessXMLRPCURL(
+        for siteAddress: String,
+        loading: @escaping ((Bool) -> ()),
+        completion: @escaping (GuessXMLRPCURLResult) -> ()
+    ) {
+        xmlrpcFacade.guessXMLRPCURL(forSite: siteAddress, success: { url in
+            // Success! We now know that we have a valid XML-RPC endpoint.
+            // At this point, we do NOT know if this is a WP.com site or a self-hosted site.
+            if let url = url {
+                self.loginFields.meta.xmlrpcURL = url as NSURL
+            }
+
+            completion(.success)
+
+            }, failure: { error in
+                guard let error = error else {
+                    return
+                }
+                // Intentionally log the attempted address on failures.
+                // It's not guaranteed to be included in the error object depending on the error.
+                WPAuthenticatorLogInfo("Error attempting to connect to site address: \(self.loginFields.siteAddress)")
+                WPAuthenticatorLogError(error.localizedDescription)
+
+                self.tracker.track(failure: .loginFailedToGuessXMLRPC)
+
+                loading(false)
+
+                guard self.isSiteDiscovery == false else {
+                    completion(.troubleshootSite)
+                    return
+                }
+
+                let err = self.originalErrorOrError(error: error as NSError)
+                self.handleGuessXMLRPCURLError(error: err, loading: loading, completion: completion)
```

Comment:
nitpick: I think this should be outdented. It would be nice if something like SwiftLint could catch this earlier.